modules/users/server/jobs/user-finish-signup.server.job.js

```diff
@@ -0,0 +1,129 @@
+    User = mongoose.model('User');
+
+// How many reminder emails we should send?
+var maxReminders = 3;
+
+// How long time between signup and first reminder?
+var firstReminderAfter = { 'hours': 4 }; // days|hours|minutes
+
+// How long time between reminders?
+var nextReminderAfter = { 'days': 2 }; // days|hours|minutes
+
+module.exports = function(job, agendaDone) {
+  async.waterfall([
+
+    // Find un-confirmed users
+    function(done) {
+
+      // Ignore very recent messages and look for only older than x seconds
+      var createdTimeAgo = new Date();
+      createdTimeAgo.setSeconds(createdTimeAgo.getSeconds() - moment.duration(firstReminderAfter).asSeconds());
+
+      // Ignore very recent messages and look for only older than x seconds
+      var remindedTimeAgo = new Date();
+      remindedTimeAgo.setSeconds(remindedTimeAgo.getSeconds() - moment.duration(nextReminderAfter).asSeconds());
+
+      User
+        .find({
+          public: false,
+          created: {
+            $lt: createdTimeAgo
+          }
+        })
+        .and([
+          {
+            $or: [
+              { publicReminderCount: { $lt: maxReminders } },
+              { publicReminderCount: { $exists: false } }
+            ]
+          },
+          {
+            $or: [
+              { publicReminderSent: { $lt: remindedTimeAgo } },
+              { publicReminderSent: { $exists: false } }
+            ]
+          }
+        ])
+        .limit(50)
+        .exec(function(err, users) {
+          done(err, users);
+        });
+
+    },
+
+    // Send emails
+    function(users, done) {
+
+      if (users.length > 0) {
+
+        // Create a queue worker to send reminders in parallel
+        // Process at most 3 reminders at the same time
+        // @link https://github.com/caolan/async#queueworker-concurrency
+        var remindersQueue = async.queue(function (user, reminderCallback) {
+
+          emailService.sendSignupEmailReminder(user, function(err) {
+            if (err) {
+              reminderCallback(err);
+            } else {
+              User.findByIdAndUpdate(
+                { _id: user._id },
+                {
+                  publicReminderSent: new Date(),
+                  $inc: { publicReminderCount: 1 }
+                },
+                reminderCallback
+              );
+            }
+          });
+
+        }, 3); // How many reminders to process simultaneously?
+
+        // Start processing reminders for all the found users
+        remindersQueue.push(users);
+
+        // Assign a final callback to work queue
+        // All jobs done, continue
+        remindersQueue.drain = function(err) {
+          if (err) {
+            console.error('Sending signup reminder mails caused an error:');
+            console.error(err);
+          }
+
+          done(null);
+        };
+      } else {
+        // No users to send emails to
+        done(null);
+      }
+
+    }
+
+  ], function(err) {
+    if (err) {
```

Comment:
can just pass `err` into `agendaDone` according to the docs [1], e.g.:

```js
agenda.define('registration email', function(job, done) {
    User.get(job.attrs.data.userId, function(err, user) {
       if(err) return done(err);
       email(user.email(), 'Thanks for registering', 'Thanks for registering ' + user.name(), done);
     });
  });
```

[1] https://github.com/rschmukler/agenda#example-project-structure